interview-questions.md

```diff
@@ -8,62 +8,105 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is a function? Why would you use one?
 
   Your answer:
+ A function is a piece of code that acts on other parts of code to do some action on my code like mathematical operation, built in methods etc.
+ I would use one because it's what actually makes my code work and do something rather than just store data.
 
   Researched answer:
 
+  Functions are reusable pieces of code, they excecute when called and always give an output, the information passed is called an argument and 
+  the function can be used as many times as needed. It has a strict syntax with name, parenthes with arguments if needed, fat arrow syntax  and curly braces with instructions inside and a return.
 
+  Functions are useful because I can define the code once, and use it many times with different arguments, to produce different results.
 
 2. What is the difference between .map() and .filter()?
 
   Your answer:
 
-  Researched answer:
+  They are both built-in methods for javascript, .map scans through an array and gives back an array with the same amount of data.
+  .filter scans through an array and not ncesessarly gives back the same amount of data, filtering out what doesn't follow the rules of the conditions of the .filter
 
+  Researched answer:
 
+  .map() and .filter() are two higher-order functions and both have three built-in arguments - the value, index, and array.
+  they both scan through an array, .map returns a new array of the same length, .filter gives a subset of the given array containing only the values that satisfy your stated condition.
 
 3. What is the difference between console.log() and return?
 
   Your answer:
+  console.log is a way in javascript to see what is happening to my code in the terminal. I can see how my data are changing and have a reference of what I'm doing or simply get a message when I run my code.
+  Return is associated to functions and it's part of its syntax, I can have only one return in a function but there's no limimtations for control.log
 
   Researched answer:
 
+  console.log() can be called as many times as we want and we can even  inside functions to see what is my output in that specific point
+  Return is part of a function syntax and it's needed to call the function, following the instructions to perform a task but itself it will not give an output in the terminal, to see the output I need to console log the function.
 
 
 4. In regards to functions, what is an argument?
 
   Your answer:
 
+ argument is the data the function is acting on.
```

Comment:
Totally. Its what we pass into the function for the function to manipulate in someway.